### ISSUE TITLE
lr-gan.pytorch update

### ISSUE THREAD

**Issue by kwon-young** [jwyang/lr-gan.pytorch#16]
Hello,
First of all, thank you for your open-source implementation of the lr-gan model.

My end goal is to build on this model in order to do unsupervised detection, and I have a few questions:

* Could you update the dead-link to the mnist1 and mnist2 dataset (or maybe publish the script you use to generate the dataset)?
* There is a discrepancy between what is explained in original paper and the actual implementation of the model relative the use of the lstm layer when generating the background.
    * In the paper, the input noise is fed to the lstm layer before being use by the background generator
    * In the implementation: [train.py:L290](https://github.com/jwyang/lr-gan.pytorch/blob/caedd45e40072b6cc4ed9a591abe8b8805373197/train.py#L290) you note that you fed the noise directly without explaining why.

I opened pull-request to contribute back and to update the code to python 3, to the most recent version of pytorch, as well as correcting some minor bugs when using the data.
I've also implemented a pytorch version of your STNM module.
Could you check if the implementation is correct?

Finally, would you be interested to actively accept pull-request concerning the use of the model for detection (or what you call conditional image generation in your paper)? The other issue where you talked about detection is quite helpful to actually implement the encoder and the reconstruction loss, but by actually implementing the architecture, I could make sure that I'm doing the correct thing.

Thank you very much for your help!
